### ISSUE TITLE
meryl crashes with "ERROR: Kmer size not supplied" when using minimap2 with canu 1.8

### ISSUE THREAD

**Issue by HenrivdGeest** [marbl/canu#1132]
Hello I am using CANU 1.8 release to assemble a plant genome with ~ 30x raw pacbio overage:
```
canu-1.8/Linux-amd64/bin/canu -version                        
Canu 1.8
```
When I supply a spec file with just my java binary location canu runs fine (tested until it runs the overlapper stage)
However, when I add the minimap parameter it crashes at the very first step with meryl:
command:
`canu-1.8/Linux-amd64/bin/canu genomeSize=xxxM -s javabin.spec -d 10cell_try11 -p 10cell_try11 -pacbio-raw something.fasta`

specfile:
```
java=/bin/java
corOverlapper=minimap
```

Crash log:
```
-- BEGIN CORRECTION
--
----------------------------------------
-- Starting command on Wed Oct 31 10:30:05 2018 with 5573.956 GB free disk space

    cd correction/0-mercounts
    ./meryl-configure.sh \
    > ./meryl-configure.err 2>&1

-- Finished on Wed Oct 31 10:30:07 2018 (2 seconds) with 5573.956 GB free disk space
----------------------------------------
--  segments   memory batches
--  -------- -------- -------

ABORT:
ABORT: Canu 1.8
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.
ABORT: Try restarting.  If that doesn't work, ask for help.
ABORT:
ABORT:   failed to parse meryl configure output 'correction/0-mercounts/.config.01.out'.
ABORT:
ABORT: Disk space available:  5573.956 GB
ABORT:
ABORT: Last 50 lines of the relevant log file (correction/0-mercounts/.config.01.out):
ABORT:
ABORT:   ERROR: Kmer size not supplied with modifier k=<kmer-size>.
ABORT:
```
Contents of the file correction/0-mercounts/meryl-configure.sh:
```
#!/bin/sh
......
#  Environment for any object storage.
export CANU_OBJECT_STORE_CLIENT=
export CANU_OBJECT_STORE_NAMESPACE=10cell_try11
export CANU_OBJECT_STORE_PROJECT=
....
....canu-1.8/Linux-amd64/bin/meryl -C k=0 threads=8 memory=24 \
  count segment=1/01 ../../10cell_try11.seqStore \
> .config.01.out 2>&1
```
full contents of .config.01.out:
```
cat 10cell_try11/correction/0-mercounts/.config.01.out
ERROR: Kmer size not supplied with modifier k=<kmer-size>.

```
Here the k parameter is set to 0 of which I doubt this is okay.
I tried setting various merSizes in my spec file, but it did not help.

**Comment by brianwalenz** [comment]
Minimap doesn't need kmers, so meryl doesn't even need to run.  Bad testing on our part!

Create the following six empty files to skip this step (the last two might not be necessary).
```
correction/0-mercounts/meryl-count.sh
correction/0-mercounts/meryl-process.sh
correction/0-mercounts/meryl-count.success
correction/0-mercounts/meryl-process.success
correction/0-mercounts/10cell_try11.dump
correction/0-mercounts/10cell_try11.ignore.gz
```

I'll get back with a code fix later.

The same failure will occur if trimming or unitigging is attempted with minimap.  That's not the default, and you don't have it enabled, but thought I'd mention for completeness.

**Comment by brianwalenz** [comment]
Fixed in the code now too.

**Comment by fbiase** [comment]
Hi Brian I am having the same problem, even though you mentioned this has been fixed. I downloaded and installed Canu on April 22nd 2019 in a server with. 
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.2 LTS
Release:        18.04
Codename:       bionic


The code I am using is
 '/home/fernando/bioinfo/canu-1.8/Linux-amd64/bin/canu  -correct  \
-p oocyte -d /data/auburn/nanopore/oocyte_pool/oocyte_pool/oocyte_fastq_corrected  \
genomeSize=2.8g cormhapMemory=98 corThreads=10 batMemory=98 batThreads=10 stopOnLowCoverage=0  minReadLength=200 minOverlapLength=200 corMhapSensitivity=high corOutCoverage=all  corOverlapper=minimap \
-nanopore-raw /data/auburn/nanopore/oocyte_pool/oocyte_pool/20190412_2052_MN24655_FAK66309_57141fb4/fastq_pass/*.fastq'

which returns me the same error mentioned above

`ABORT:
ABORT: Canu 1.8
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.
ABORT: Try restarting.  If that doesn't work, ask for help.
ABORT:
ABORT:   failed to parse meryl configure output 'correction/0-mercounts/.config.01.out'.
ABORT:
ABORT: Disk space available:  7018.29 GB
ABORT:
ABORT: Last 50 lines of the relevant log file (correction/0-mercounts/.config.01.out):
ABORT:
ABORT:   ERROR: Kmer size not supplied with modifier k=<kmer-size>.
`

**Comment by skoren** [comment]
This isn't in the release, you'd have to get the latest from the repo and compile yourself to get the fix.
